### PR TITLE
coordinator: fix wrong policy route when there is more than 1 secondary nics

### PR DESCRIPTION
**Pull request by cyclinder**
#### What type of PR is this?

- release/bug 

**What this PR does / why we need it**:

- No longer rely on the API to get the rule number of the NIC, getting it through the API may cause the number to be wrong, which can avoid the wrong policy route when there is more than 1 secondary nics.
- New a netlink.Route object without use of raw object when adding the route, which can avoid netlink return "invalid argument" error.

**Which issue(s) this PR fixes**:

Fixes https://github.com/spidernet-io/spiderpool/issues/3877

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/networking/networking/route.go [hunk 1]
```diff
@@ -257,10 +258,11 @@ func moveRouteTable(linkIndex, srcRuleTable, dstRuleTable int, onlyCopyOverlayDe
 		}
 
 		// we need copy the all routes in main table of the podDefaultRouteNic to dstRuleTable.
-		// Otherwise, the reply packet don't know
+		// Otherwise, we don't know how to forward the packet send from the nic
+		route.Table = dstRuleTable
 		if err = netlink.RouteAdd(&route); err != nil && !os.IsExist(err) {
-			logger.Error("failed to RouteAdd in new table ", zap.String("route", route.String()), zap.Error(err))
```
Comment (weizhoublue) [inline pkg/networking/networking/route.go:262]: route netlink.Route  进行 copy 修改，避免潜在 bug
相关日志也把 要生效的最终路由 打下

**Comment by ty-dc** [comment]
Can we add an issue?

File: pkg/networking/networking/route.go (hunk 1 above)
Comment (cyclinder) [reply pkg/networking/networking/route.go:262]: Done

**Comment by weizhoublue** [comment]
need detailed description in the PR to explain the bug's influence , for release note

**Comment by weizhoublue** [comment]
相关 文档，提到了 pod 内 

# ip route show table 500

File: cmd/coordinator/cmd/utils.go [hunk 2]
```diff
@@ -69,27 +58,45 @@ func (c *coordinator) autoModeToSpecificMode(mode Mode, podFirstInterface string
...
-			}
-
-			if !exist {
-				return fmt.Errorf("when creating interface %s in underlay mode, it detects that the auxiliary interface %s was not created by previous interface. please enable coordinator plugin in previous interface", c.currentInterface, podFirstInterface)
-			}
+		if !c.firstInvoke && !vethExist {
+			return fmt.Errorf("when creating interface %s in underlay mode, it detects that the auxiliary interface %s was not created by previous interface. please enable coordinator plugin in previous interface", c.currentInterface, podFirstInterface)
+		}
+
+		// eth0, net1 ==> 101
```
Comment (weizhoublue) [inline cmd/coordinator/cmd/utils.go:90]: 这些 comment 正式一点

**Comment by cyclinder** [comment]
wait for https://github.com/spidernet-io/spiderpool/pull/3906

File: cmd/coordinator/cmd/utils.go (hunk 2 above)
Comment (cyclinder) [reply cmd/coordinator/cmd/utils.go]: done

**Review by ty-dc — APPROVED** [review]

File: test/e2e/reclaim/reclaim_test.go [hunk 3]
```diff
@@ -587,7 +587,7 @@ var _ = Describe("test ip with reclaim ip case", Label("reclaim"), func() {
 			Expect(frame.DeletePod(podName, namespace)).To(Succeed(), "Failed to delete pod %v/%v\n", namespace, podName)
 			GinkgoWriter.Printf("succeed to delete pod %v/%v\n", namespace, podName)
 		},
-			Entry("a dirty IP record (pod name is wrong or containerID is wrong) in the IPPool should be auto clean by Spiderpool", Serial, Label("G00005", "G00007")),
+			PEntry("a dirty IP record (pod name is wrong or containerID is wrong) in the IPPool should be auto clean by Spiderpool", Serial, Label("G00005", "G00007")),
```
Comment (weizhoublue) [inline test/e2e/reclaim/reclaim_test.go:590]: @ty-dc 该ci是否已修复？ 后续打开

**Review by weizhoublue — APPROVED** [review]

**Pull request merged**